### PR TITLE
Show public messages

### LINKED ISSUE #71: Show Public Messages
We've talked a few times about rendering public messages for routes that are servicing the chosen stop(s). A consideration is spacing... there can be multiple messages (up to 255 characters each) and we often display multiple stops. Perhaps we could have a flag to:

1. Show departures
2. Show messages
3. Show both (default)

Other notes: 
* We used to do this in the [decommissioned](https://github.com/umts/BusInfoBoard/pull/44) interactive mobile version (though it required the user to tap an icon).
* We briefly discussed trying to figure out which routes service a stop (in case a route isn't servicing it in the next 2-3 hours), but we think the overhead might be too great with [Stops_GetAllStopsForRoutes](
https://bustracker.pvta.com/InfoPoint/swagger/ui/index#!/Stops/Stops_GetAllStopsForRoutes), and anyway just showing messages for currently displayed routes (as well as general messages) ought to be fine.

@dfaulken is working on it.

### PR DESCRIPTION
**Pull request by dfaulken**
Closes #71.

Default view:
screen shot 2017-12-04 at 11 45 38 am

Messages only:
screen shot 2017-12-04 at 11 54 47 am

No messages:
screen shot 2017-12-04 at 11 54 54 am

### CODE REVIEW

**Comment by dfaulken** [comment]
So before we merge this, we should contact the organizations using the various kiosks. If they want to control when they start displaying public messages, in case they want to monitor how that change looks on their kiosk, they should set `&show_messages=no` now. It will have no effect on the current master codebase, and will cause there to be no change once this is merged.

File: js/main.js [hunk 1]
```diff
@@ -35,11 +35,13 @@ var options =
   end_animation: 'fadeOut',       // default animate CSS for everything to be removed with at once
   alternateInterval: 3000,        // default time in ms between alternating time and interval
   disableAlternation: false,      // whether alternation should be disabled
+  showMessages: 'yes',            // whether to show messages - 'yes', 'no', or 'only' for no departures
```
Comment (dfaulken) [inline js/main.js]: Truth be told, it doesn't need to be `'yes'`. It could be anything that isn't `'no'` or `'only'` (including not defined).

**Comment by dfaulken** [comment]
Thanks @Anbranin. I'll wait for @werebus as well on this one (unless we think it's particularly pressing).

**Comment by sherson** [comment]
> So before we merge this, we should contact the organizations using the various kiosks. If they want to control when they start displaying public messages, in case they want to monitor how that change looks on their kiosk, they should set `&show_messages=no` now. It will have no effect on the current master codebase, and will cause there to be no change once this is merged.

Yesterday, I reached out to folks at the GRC and the Design Building about that.

> 2. In the case of your screenshots, the _text_ of the alert contains the routes affected. That's pretty buch how we always do it, because we're copy-pasting from Twitter and Facebook, but do we want to rely on that? Should we show the affected routes somehow?

Good point. Perhaps we can leave the routes out messages, and prefix them. For instance, currently the feed has:

``` XML
  <PublicMessage>
    <ChannelMessages />
    <DaysOfWeek>Monday Tuesday Wednesday Thursday Friday Saturday Sunday</DaysOfWeek>
    <FromDate>2017-07-31T00:00:00</FromDate>
    <FromTime>2017-07-31T00:00:00</FromTime>
    <Message>The Northampton Lodging bus stop is relocated 100-feet Southbound due to construction and sidewalk closures at the Northampton Lodging.</Message>
    <MessageId>30010</MessageId>
    <MessageTranslations />
    <Priority>Low</Priority>
    <PublicAccess>19</PublicAccess>
    <Published>true</Published>
    <Routes xmlns:d3p1="http://schemas.microsoft.com/2003/10/Serialization/Arrays">
      <d3p1:int>30041</d3p1:int>
      <d3p1:int>30042</d3p1:int>
      <d3p1:int>30048</d3p1:int>
    </Routes>
    <Signs xmlns:d3p1="http://schemas.microsoft.com/2003/10/Serialization/Arrays">
      <d3p1:int>999</d3p1:int>
    </Signs>
    <ToDate>2018-08-31T00:00:00</ToDate>
    <ToTime>2017-07-31T23:59:59</ToTime>
  </PublicMessage>
```

That'd look fine as _"Routes 41, 42, 48: The Northampton Lodging bus stop is relocated 100-feet Southbound due to construction and sidewalk closures at the Northampton Lodging."_

Potentially complicating things is that there will be a few different organizations updating this. I'll see if the others are amenable to adopting such a convention. 

Another aspect to this is that Avail supports tweeting PublicMessages automatically. For various reasons, we do not currently do this with [@umasstransit](https://twitter.com/umasstransit) or [@pvta](https://twitter.com/pvta). I'll find out if such tweets can include the routes (in case PVTA wants to start using this in the future), but this doesn't block us from deploying.

> 3.  What about non-route-specific alerts. Should we handle those?

We should render those along with messages corresponding to routes in the list of departures.

**Comment by dfaulken** [comment]
Showing route names with alerts:

screen shot 2017-12-07 at 8 40 17 am

Showing multiple alerts:

screen shot 2017-12-07 at 8 44 30 am

**Comment by dfaulken** [comment]
@sherson do you think we're all set to merge this?

**Comment by sherson** [comment]
Yep

**Comment by sherson** [comment]
Looks good 

businfoboard with public messages

**Pull request merged**